src/Rings/mpolyquo-localizations.jl

```diff
@@ -2407,6 +2407,14 @@ function Base.:(==)(
   return all(x->f(x) == g(x), gens(domain(f)))
 end
 
+function Base.hash(f::Map{<:MPolyAnyRing, <:MPolyAnyRing}, h::UInt)
+    h = hash(domain(f), h)
+    h = hash(codomain(f), h)
+    # TODO: add in coefficient_map if available
```

Comment:
```suggestion
    _has_coefficient_map(f) && (h = hash(coefficient_map(f), h))
```
Problem is: coefficient maps can be anything. In particular they can be mathematically the same, even though they have different data types in the computer and, hence, most probably also different hashes. So one could also think about restricting the hash and comparison functions to maps without coefficient maps and throw an error.

Reply:
I saw `_has_coefficient_map` and did not add code like in your suggestion precisely because of the issue you point out. 

I guess could compute the image of the generators under the coefficient map and hash that; but it is not clear to me this will work in general, and also the performance impact (this will cause a bunch of allocations!) is unclear.

Of course we could cache the hash (rhyming not intended ;-) ) to overcome that.

Reply:
But note that this is strictly optional: if we just don't consider the coefficient maps for the hash, we still have something that is correct.

Reply:
> but it is not clear to me this will work in general

No, it won't. Take, for instance, the map on `(QQ[i])[x, y]` which is the identity on the variables, but complex conjugation on the coefficients.

Reply:
> if we just don't consider the coefficient maps for the hash, we still have something that is correct.

True. So hashing could (and probably should) ignore the coefficient map. The equality check, on the other hand, should throw an error in case there are coefficient maps present which are not `===`. In that case I think, we simply can not write generic code due to the generality allowed for coefficient maps. 

@thofma : Am I seeing this correct?

Reply:
Yes, and yes to throwing an error.

(Maybe just do `hash(f::Map, h::UInt) = UInt(0)` and be done with it.)

Reply:
Error: fine by me.

`hash(f::Map, h::UInt) = UInt(0)`:  this discards any previous hash results, which is bad when chaining hashes; better would be `hash(f::Map, h::UInt) = h` . That said, I personally prefer an `error` in this case, i.e. `hash(f::Map, h::UInt) = error("hashing not implemented for this map type, please report this as bug to the OSCAR team")` :-)